src/qtcore/qml/qml.js

```diff
@@ -306,6 +308,11 @@ function applyProperties(metaObject, item, objectScope, componentScope) {
     _executionContext = componentScope;
     for (i in metaObject) {
         var value = metaObject[i];
+        if (i == "id" || i == "$class") { // keep them
+          item[i] = value;
+          continue;
+        }
```

Comment:
@pavelvasev, what's going on here? This contradicts the check below, `i == 'id'` below never fires.